atlite/wind.py

```diff
@@ -90,3 +101,105 @@ def extrapolate_wind_speed(ds, to_height, from_height=None):
                           "units" : "m s**-1"})
 
     return wnd_spd.rename(to_name)
+
+def download_turbineconf(turbine=None, store_locally=True):
+    """Download a windturbine configuration from the OEDB database.
+    
+    Download the configuration of a windturbine model from the OEDB database
+    into the local 'windturbine_dir'.
+    The OEDB database can be viewed here:
+    https://openenergy-platform.org/dataedit/view/supply/turbine_library
+    (2019-07-22)
+    Only one turbine configuration is downloaded at a time, if the 
+    search parameters yield an ambigious result, not data is downloaded.
+    
+    Parameters
+    ----------
+    turbine : dict
```

Comment:
```suggestion
    turbine : dict|str
```

Let's also allow a *smart* string:
```python
if isinstance(turbine, str):
    try:
        turbine = {'id': int(turbine)}
    except ValueError:
        turbine = {'name': turbine}
```

this would only fail if the name is not specific enough that one has to specify a manufacturer, as well; and work fine otherwise.

Reply:
I see the appeal, but I am against it limiting ourselves then only to id/name/manu. .
What would you say if we allow for any field from the OEDB and if the provided data is insufficient for a unique retrieve we fail loud?

Reply:
It's not limiting if it is mainly intended as an additional convenience feature.

I'd allow:
```python
# As convencience all equivalent
download_turbineconf(13)
download_turbineconf({'id':13})
# Maybe as well: download_turbineconf(id=13)

download_turbineconf('E-53/800')
download_turbineconf({'name': 'E-53/800'})
# Maybe as well: download_turbineconf(name='E-53/800')

download_turbineconf('Enercon_E-53/800') # not incidentally exactly the name the turbine would be saved to (except for replacements)
download_turbineconf({'manufacturer': 'Enercon', 'name': 'E-53/800'})
# Maybe as well: download_turbineconf(manufacturer='Enercon', name='E-53/800')
```

I'd gladly support all fields in the second or third manner, but would reserve the simple string/int for `id`, `name` and `manufacturer`/`name`.